### PR TITLE
Feat/login page

### PR DESCRIPTION
**Pull request by tyleryy**
### Overview
Login page for hackers to login and apply for IrvineHacks. For Issue #81. You do not need to review the components in the `components/ui` folder of the NextJS site project folder. These are autogenerated shadcn components that were installed. Refer to the docs for more info: https://ui.shadcn.com/docs.

### CODE REVIEW

**Comment by tyleryy** [comment]
Uses shadcn, so #80 is a parent PR to this one.

File: apps/site/src/app/login/page.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+export const revalidate = 60;
+
+const LOGIN_PATH = "/api/user/login";
+
+const formSchema = z.object({
+	email: z
+		.string()
+		.email({ message: "Sorry, that email address is invalid." })
+		.refine((email) => email.endsWith(".edu"), {
```
Comment (rileysw) [inline apps/site/src/app/login/page.tsx:28]: @taesungh @samderanova Are we okay removing .edu validation to allow high school students to sign up?

Probably also need to add a message saying to use school email if you have one

**Comment by IanWearsHat** [comment]
The footer for the login page seems to be floating off the bottom for me

**Comment by tyleryy** [comment]
> The footer for the login page seems to be floating off the bottom for me

oh oops lol i'll fix that

**Comment by rileysw** [comment]
Also, any reason why the button is light blue?

**Comment by IanWearsHat** [comment]
Will mobile responsiveness be in a future issue?

**Comment by tyleryy** [comment]
I just made it blue because. I can change it if you would like

**Comment by tyleryy** [comment]
Is it not responsive? I thought I made it so. If not send a screenshot I'll fix it

**Comment by rileysw** [comment]
> I just made it blue because. I can change it if you would like

Probably the gold button for consistency

**Comment by rileysw** [comment]
> > I just made it blue because. I can change it if you would like
> 
> Probably the gold button for consistency
> 
> Oh it's not a white background...I don't have a better suggestion, that's a graphics thing 😢

On second thought, can you use white background? At least that's what the Figma indicates

**Comment by tyleryy** [comment]
> > > I just made it blue because. I can change it if you would like
> > 
> > 
> > Probably the gold button for consistency
> > Oh it's not a white background...I don't have a better suggestion, that's a graphics thing 😢
> 
> On second thought, can you use white background? At least that's what the Figma indicates

Yea in my personal opinion I didn't like the color scheme on the Figma design. Mine isn't great either, but I just wanted to get functionality working first and then change colors later. We can talk to graphics about it.

**Comment by waalbert** [comment]
It looks like this includes the commits from PR [#80 ](https://github.com/HackAtUCI/irvinehacks-site-2024/pull/80)

**Comment by taesungh** [comment]
Agree with Albert here, if we need to set up shadcn first, let's have that in a separate PR and contain the scope of subsequent PRs.

**Comment by tyleryy** [comment]
Yea thats what #80 is supposed to be.

**Comment by taesungh** [comment]
We need the login page long before the schedule, so we should extract what's needed for both and review that before reviewing the login page design.

**Comment by tyleryy** [comment]
> I think what they're saying is that we should fix up and merge 80 before we work on this because 80 isn't done yet. I'm assuming you based this PR off of 80?

Yea I did because 80 is where I set up shadcn.

**Comment by samderanova** [comment]
Can you extract only the stuff we need for the login page (shadcn and components) and then keep it here instead of basing this PR off of #80 ?

**Comment by taesungh** [comment]
We are missing the schedule page now. Could you rebase/cherry pick only the relevant changes as originally suggested?

**Comment by samderanova** [comment]
Closing this as completed by #114 .

**Pull request closed without merging**